src/App.tsx

```diff
@@ -1,73 +1,103 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './components/DropdownMenu';
+import { Person } from './types/Person';
+import cn from 'classnames';
+import '@fortawesome/fontawesome-free/css/all.css';
+import debounce from 'lodash.debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const filterUsers = useMemo(() => {
+    const filteredUsers = peopleFromServer.filter(user =>
+      user.name.toUpperCase().includes(appliedQuery.toUpperCase()),
+    );
+
+    setHasError(filteredUsers.length === 0);
+
+    return filteredUsers;
+  }, [appliedQuery]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const handleInputClick = () => {
+    setIsDropdownActive(true);
+  };
+
+  const reset = () => {
+    setQuery('');
+    setSelectedPerson(null);
+    setAppliedQuery('');
+  };
+
+  const handleUserSelect = (user: Person) => {
+    setSelectedPerson(user);
+    setQuery(user.name);
+    setIsDropdownActive(false);
+  };
+
+  const title = selectedPerson
+    ? `${selectedPerson?.name} (${selectedPerson?.born} - ${selectedPerson?.died})`
+    : 'No selected person';
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {title}
         </h1>
 
-        <div className="dropdown is-active">
+        <div
+          className={cn('dropdown', { 'is-active': isDropdownActive })}
+          onMouseLeave={() => setIsDropdownActive(false)}
+        >
           <div className="dropdown-trigger">
             <input
+              value={query}
+              onChange={handleChange}
+              onClick={handleInputClick}
```

Comment:
```suggestion
              onFocus={handleInputClick}
```
This is the reason of failing test